### PR TITLE
Add MediaSetAccess::provideFileFromUrl

### PR DESCRIPTION
**Pull request by bzeller**

### CODE REVIEW

File: zypp/PublicKey.h [hunk 1]
```diff
@@ -331,8 +334,6 @@ namespace zypp
     friend class KeyRing;
     /** KeyRing ctor: No need to parse file if KeyRing already had valid KeyData. */
     PublicKey( const filesystem::TmpFile & sharedFile_r, const PublicKeyData & keyData_r );
-    /** KeyRing ctor: Legacy callback APIs take PublicKey, but just need the PublicKeyData No need to export to file. */
-    explicit PublicKey( const PublicKeyData & keyData_r );
```
Comment (mlandres) [inline zypp/PublicKey.h]: Wanted to cherry-pick this commit, but PublicKey.h change is not related (if necessary at all).

Comment (bzeller) [reply zypp/PublicKey.h]: That was a leftover from my tests with loading all keys manually with KeyManager and trying to register them manually. Cleaned the branch up to only contain related changes
